### PR TITLE
feat: add possibility to change vote

### LINKED ISSUE #874: Add the ability to change your vote
While the voting period is open, it's possible to change your vote via the smart contracts. We should add that ability to the UI, pretty much identical to how we've handled it on Omm:

Screen Shot 2022-02-02 at 2 59 54 PM
Screen Shot 2022-02-02 at 3 00 15 PM

@hetfly are you able to adapt this for the Balanced UI? Happy to provide more design guidance if needed.

### PR DESCRIPTION
**Pull request by hetfly**
close #874

### CODE REVIEW

File: src/app/components/ProposalModal/index.tsx [hunk 1]
```diff
@@ -64,9 +66,11 @@ export function ProposalModal(props: ProposalProps) {
     <Modal isOpen={isOpen} onDismiss={onCancel}>
       <Flex flexDirection="column" alignItems="stretch" m={5} width="100%">
         <Typography variant="content" textAlign="center" mb={1}>
-          Submit vote?
+          {UIStatus === ModalStatus.ChangeToApprove || UIStatus === ModalStatus.ChangeToReject
+            ? 'Change Vote?'
```
Comment (parrot9design) [inline src/app/components/ProposalModal/index.tsx:70]:
```suggestion
            ? 'Change vote?'
```

File: src/app/components/ProposalModal/index.tsx [hunk 2]
```diff
@@ -99,7 +103,11 @@ export function ProposalModal(props: ProposalProps) {
           {!shouldLedgerSign && (
             <>
               <CancelButton onClick={onCancel}>Cancel</CancelButton>
-              <SubmitButton onClick={onSubmit}>Submit vote</SubmitButton>
+              <SubmitButton onClick={onSubmit}>
+                {UIStatus === ModalStatus.ChangeToApprove || UIStatus === ModalStatus.ChangeToReject
+                  ? 'Change Vote'
```
Comment (parrot9design) [inline src/app/components/ProposalModal/index.tsx:108]:
```suggestion
                  ? 'Change vote'
```

File: src/app/components/ProposalModal/index.tsx [hunk 3]
```diff
@@ -99,7 +103,11 @@ export function ProposalModal(props: ProposalProps) {
...
+                  : 'Submit Vote'}
```
Comment (parrot9design) [inline src/app/components/ProposalModal/index.tsx:109]:
```suggestion
                  : 'Submit vote'}
```

**Review by parrot9design — COMMENTED** [review]
This is great, thanks @hetfly! I've just suggested a couple of capitalisation changes, but otherwise this works as expected.

**Review by parrot9design — APPROVED** [review]

**Comment by octocat512** [comment]
Thanks @hetfly

**Pull request merged**